pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -404,7 +431,25 @@ async def _prepare_request(
         # Increment run_step
         ctx.state.run_step += 1
 
-        model_settings = merge_model_settings(ctx.deps.model_settings, None)
+        # Fix max_tokens
+        patch_model_settings: ModelSettings = {}
+        if (
+            ctx.deps.model_settings
+            and ctx.deps.model_settings.get('max_tokens')
```

Comment:
```suggestion
            and (max_tokens := ctx.deps.model_settings.get('max_tokens'))
```

This way we can drop the extra variable below